edb/server/compiler/compiler.py

```diff
@@ -152,7 +153,12 @@ def compile_edgeql_script(
     compiler._std_schema = std_schema
     compiler._bootstrap_mode = True
 
+    # Note: there is a lot of GC activity due to contexts and other
+    # Schema-related reference cycles.  Postponing collection until after
+    # the compilation is complete speeds it up between 7% - 25%.
+    gc.disable()
     units = compiler._compile(ctx=ctx, eql=eql.encode())
+    gc.enable()
```

Comment:
What do you think about also calling `gc.collect()` here?

Reply:
54% of compilations during benchmarking and profiling don't end up reaching the first generation GC threshold anyway. 78% don't reach the threshold for older generations.
So, collecting always, while more predictable, would slow things down, especially for smaller queries.

Reply:
OK, let's not do it then.